crypto/fipsmodule/ec/p521.c

```diff
@@ -683,53 +775,53 @@ static void ec_GFp_nistp521_point_mul(const EC_GROUP *group, EC_RAW_POINT *r,
+// The |p521_g_pre_comp| table has 27 sub-tables each holding 16 points:
+//      0 :       [1]G,       [3]G,  ...,       [31]G
+//      1 :  [1*2^20]G,  [3*2^20]G,  ...,  [31*2^20]G
 //                         ...
-//      i : [1*2^28i]G, [3*2^28i]G,  ..., [127*2^28i]G
+//      i : [1*2^20i]G, [3*2^20i]G,  ..., [31*2^20i]G
 //                         ...
-//     18 :   [2^504]G, [3*2^504]G,  ..., [127*2^504]G
+//     26 :   [2^520]G, [3*2^520]G,  ..., [31*2^520]G
 // Computing the negation of a point P = (x, y) is relatively easy:
 //     -P = (x, -y).
-// So we may assume that for each sub-table we have 128 points instead of 64:
-//     [\pm 1*2^28i]G, [\pm 3*2^28i]G, ..., [\pm 127*2^28i]G.
+// So we may assume that for each sub-table we have 32 points instead of 16:
+//     [\pm 1*2^20i]G, [\pm 3*2^20i]G, ..., [\pm 31*2^20i]G.
 //
-// The 521-bit |scalar| is recoded (regular-wNAF encoding) into 75 signed
-// digits, each of length 7 bits, as explained in the
+// The 521-bit |scalar| is recoded (regular-wNAF encoding) into 105 signed
+// digits, each of length 5 bits, as explained in the
 // |p521_felem_mul_scalar_rwnaf| function. Namely,
-//     scalar' = s_0 + s_1*2^7 + s_2*2^14 + ... + s_74*2^518,
-// where digits s_i are in [\pm 1, \pm 3, ..., \pm 127]. Note that for an odd
+//     scalar' = s_0 + s_1*2^5 + s_2*2^10 + ... + s_104*2^520,
+// where digits s_i are in [\pm 1, \pm 3, ..., \pm 31]. Note that for an odd
 // scalar we have that scalar = scalar', while in the case of an even
 // scalar we have that scalar = scalar' - 1.
 //
 // To compute the required product, [scalar]G, we may do the following.
 // Group the recoded digits of the scalar in 4 groups:
-//                                           |   corresponding multiples in
-//                   digits                  |   the recoded representation
-//    -------------------------------------------------------------------------
-//    (0): {s_0, s_4,  s_8, ..., s_68, s_72} |  { 2^0, 2^28, ...,      , 2^504}
-//    (1): {s_1, s_5,  s_9, ..., s_69, s_73} |  { 2^7, 2^35, ...,      , 2^511}
-//    (2): {s_2, s_6, s_10, ..., s_70, s_74} |  {2^14, 2^42, ...,      , 2^518}
-//    (3): {s_3, s_7, s_11, ..., s_71}       |  {2^21, 2^49, ..., 2^497}
-//         corresponding sub-table lookup    |  {  T0,   T1, ...,   T17,   T18}
+//                                            |   corresponding multiples in
+//                  digits                    |   the recoded representation
+//   -------------------------------------------------------------------------
+//   (0): {s_0, s_4,  s_8, ..., s_100, s_104} |  { 2^0, 2^20, ..., 2^500, 2^520}
+//   (1): {s_1, s_5,  s_9, ..., s_101}        |  { 2^5, 2^25, ..., 2^505}
+//   (2): {s_2, s_6, s_10, ..., s_102}        |  {2^10, 2^30, ..., 2^510}
+//   (3): {s_3, s_7, s_11, ..., s_103}        |  {2^15, 2^35, ..., 2^515}
+//        corresponding sub-table lookup      |  {  T0,   T1, ...,   T17,   T18}
```

Comment:
number of tables is 27, so the last ones are T25, T26

Reply:
done